libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -216,6 +216,14 @@ const AP_Param::GroupInfo AP_GPS::var_info[] = {
     AP_GROUPEND
 };
 
+// constructor
+AP_GPS::AP_GPS(GPS_Engine_Setting nav_filter_default)
+{
+    AP_Param::setup_object_defaults(this, var_info);
+
+    _navfilter.set_default(nav_filter_default);
```

Comment:
Since this is a single setting, why don't vehicles use the param method to set the default value? It would be much simpler and doesn't force to change every user of GPS library.